### DISCUSSION TITLE
Essential inquiry to install flowpilot

### DISCUSSION THREAD

**Discussion by iamshreeram** [flowdriveai/flowpilot#68] [General]
Hello, I'm interested in testing out the flow pilot in my Honda Civic '19 and I'd like to select the appropriate hardware for it. I have a few inquiries.

1. Is it possible to use Flow Pilot with android phones other than One plus 6T / 7T and Google Pixel models listed on the [GitHub page's compatibility list](https://github.com/flowdriveai/flowpilot/wiki/Phone-Compatibility-List)? Does it imply that other android devices need extra configurations in contrast to the compatible ones? Can Flow Pilot be utilized with Samsung or Google Pixel?

2. Which phone is the best recommendation for running the functionality of Flow Pilot effectively? Does the OnePlus 7T offer a wide camera or does it require better hardware performance than 7T?

3. In the [instructions](https://github.com/phr00t/flowpilot/wiki/How-to-make-an-LG-G8-Flow-Pilot-Device) provided by @phr00t for making an LG G8 Flow Pilot device, there are hardware modifications mentioned from step 7 to step 12. Are these modifications specific to LG phones or are they necessary for any phone to run stably?

4. Is there a paid membership or is the installation process different for premium version of flow pilot? How can someone access the premium version of the Flow Pilot app?

5. Is it correct that the black/red-colored car harness can be used for PC, laptop, and Android phones? Additionally, when should each colored panda be used?

6. Is the giraffe only required for the white or grey panda?

7. Can I follow the [simplified Flow Pilot install process mentioned in this link](https://github.com/phr00t/flowpilot/wiki/Simplified-Flow-Pilot-Install-Process) for the software installation? It seems to be a straightforward process.

**Questions w.r.t Comma** : 

8. Can you provide a list of the items to purchase from comma? For example, 
- Compatible Panda - White, Grey, Red, Black
- Giraffe 
- Harness Connector 
- Car ODB 2 port to comma power 

9. Is it possible to use GPS or map navigation in Flow Pilot, similar to how it is used in Comma?

10. Comma stores user driving history and monitors user awareness through the front camera and steering wheel at regular intervals. Does Flow Pilot have similar features?

11. Is it possible to use Comma3 (premium) in India with just a harness, without the need for a panda?

**Comment by iamshreeram** [comment]
12. Does the Flow pilot feature function universally in all countries, or does it require configuration to select the appropriate lane based on the country?
13. Which version of Openpilot is compatible with Flowpilot? Can I update to newer openpilot versions when comma releases them, or is it recommended to stick with the version I initially install?

15. What is the process of determining the correct car harness for Indian cars?

16. Is it necessary to have all the items listed below to purchase from Comma? 

- Panda : https://comma.ai/shop/panda
- Harness connector : https://comma.ai/shop/harness-connector
- Harness : https://comma.ai/shop/car-harness
- Harness box : https://comma.ai/shop/harness-box

17. Btw, do these belong exclusively to Comma? Aside from panda, can we create the harness, harness connector, and harness box on our own with 3d printer? Bcz, If any of these small connectors does not fit the Indian car, shipping from the US each time would pose a significant challenge.


@MankaranSingh , Appreciate if you could review the above inquiries when it is convenient for you. Thank you.

**Comment by MankaranSingh** [comment]
Phroot's fork is targeted to lg devices. Follow these instructions https://github.com/flowdriveai/flowpilot/wiki/Installation. The original flowpilot should work on any android phone with opencl support (majority). 

For panda, if your car is CAN-FD then use red panda, else black panda is enough. 

Does your honda civic have adas inbuilt ?

**Reply by iamshreeram** [reply]
@MankaranSingh , Thank you for investigating this.
1. How can I determine if a car has CAN-FD or CAN? Can the manufacturer provide this information or can it be found on their website?
2. Yes, the Honda Civic has built-in ADAS. However, my intention is not limited to just my own car. I would like to introduce Flowpilot to my neighborhood and make it accessible to a larger population in India. The introductory video for the Alto has sparked my interest, but it lacks specific details about the modifications made to the actuators or the methods used to achieve them.

In order to bring Flowpilot to cars in India, we need a straightforward approach, particularly for highly popular and frequently sold models (similar to what OpenPilot offers, [as shown on their website](https://comma.ai/vehicles)). By providing the car name and model, people can easily purchase the Comm3x and the necessary harness. It would be beneficial if Flowpilot could recommend the required components based on car details and model.

Having a classification or list of cars supported and not supported by Flowpilot, taking into account various features such as CAN-FD or CAN, ADAS, LKAS, EPS, etc., would be advantageous. This would particularly benefit the most sold cars in India, such as Maruti Suzuki Alto, Maruti Suzuki Swift, Maruti Suzuki WagonR, Maruti Suzuki Ertiga, Maruti Suzuki Baleno, Maruti Suzuki Brezza, Tata Nexon, Maruti Suzuki Dzire, Hyundai Creta, and Tata Punch.

**Comment by phr00t** [comment]
@iamshreeram see this FAQ: https://github.com/phr00t/flowpilot/wiki/FlowPilot:-FAQ

**Comment by iamshreeram** [comment]
@phr00t , Thank you for the FAQ. I am compiling the answers for above questions, based on discussions in discord and other sources. Please let me know if any information provided here is not relevant. 

1. Flow Pilot can be used with any android phones, although it has been specifically tested and works seamlessly with Oneplus 6T/7T. However, for other phones, some adjustments may be required to ensure that Flow Pilot functions properly. It is essential to check if the phone supports SNPE Snapdragon 855.

2. The recommended phones for use with Flow Pilot are OnePlus 6T and 7T. 

3. ?? 

4. Previously, Flowpilot had a separate branch called Premium, which included optimization and experimental features. However, this branch has been merged into the main branch or can be accessed through phr00t's version of Flow Pilot. 

5. If your car supports CAN-FD, you can use the RED panda. Otherwise, the BLACK panda is sufficient. 

6. Giraffe is not necessary for Android phones. According to [Flowpilot's wiki](https://github.com/flowdriveai/flowpilot/wiki/Connecting-to-Car), Giraffe is only required for white or grey pandas. It is unclear why someone would use white/grey pandas.  

7. phr00t's version of Flow Pilot includes the installation step with tar. This version provides the latest model and may require some advanced installation steps. The default installation of Flow Pilot may work with outdated models. 

8. Giraffe is not required for Android phones, but other components are necessary. 

9. ??

10. ??

11. ??

12. Flow Pilot should work universally, regardless of whether the car is right-hand driving or left-hand driving. However, I need to check, how the car decides which lane to choose when multiple lanes are available for single side traffic. 

13. Flow Pilot does not have a specific version of Openpilot. According to **phr00t**, many updates from Openpilot are not relevant to Flow Pilot. If there are updates that are useful for a specific car, users would need to merge those updates themselves. Following the official Flow Pilot instructions may result in outdated software with limited upgrade potential. 

14. phr00t is unsure if a compatible harness exists. If it is not available, it may be necessary to build one. 

15. Yes, the Car Harness Kit includes a car harness and a harness connector specifically designed for the chosen car. 

16. phr00t suggests that it is possible a compatible harness already exists, but it may be necessary to compare the connector with existing harness wire diagrams to confirm compatibility.

**Comment by iamshreeram** [comment]
@MankaranSingh / @phr00t -
17. Similiar to comma prime / comma sim, If a SIM card is inserted into an Android device with flowpilot installed, would it be possible to establish remote SSH access to the terminal? Can commands be sent remotely to control my car?
